pkg/controller/statusmanager/status_manager.go

```diff
@@ -65,30 +68,94 @@ func (c ClusteredName) String() string {
 	return c.ClusterName + string(ClusteredNameSeparator) + c.Namespace + string(ClusteredNameSeparator) + c.Name
 }
 
+const generateStatusSelector = names.GenerateStatusLabel // Just a single label is an EXISTS selector. Easy.
+
 // StatusManager coordinates changes to ClusterOperator.Status
 type StatusManager struct {
 	sync.Mutex
 
-	client cnoclient.Client
-	name   string
+	client           cnoclient.Client
+	name             string
+	hyperShiftConfig *network.HyperShiftConfig
 
 	failing         [maxStatusLevel]*operv1.OperatorCondition
 	installComplete bool
 
-	daemonSets     []ClusteredName
-	deployments    []ClusteredName
-	statefulSets   []ClusteredName
-	relatedObjects []configv1.ObjectReference
+	dsInformer cache.SharedIndexInformer
+	dsLister   DaemonSetLister
 
-	hyperShiftConfig *network.HyperShiftConfig
+	depInformer cache.SharedIndexInformer
+	depLister   DeploymentLister
+
+	// StatefulSets may exist in more than one cluster (in the case of Hypershift)
+	ssInformers map[string]cache.SharedIndexInformer
+	ssListers   map[string]StatefulSetLister
+
+	relatedObjects []configv1.ObjectReference
 }
 
 func New(client cnoclient.Client, name string) *StatusManager {
-	return &StatusManager{
+	s := &StatusManager{
 		client:           client,
 		name:             name,
 		hyperShiftConfig: network.NewHyperShiftConfig(),
 	}
+	s.initInformers()
+	return s
+}
+
+func (s *StatusManager) initInformers() {
+	s.dsInformer = v1appsinformers.NewFilteredDaemonSetInformer(
+		s.client.Default().Kubernetes(),
+		"", // all namespaces
```

Comment:
`metav1.NamespaceAll`
(and again below)

Reply:
Good point.